substrate/frame/staking/src/ledger.rs

```diff
@@ -201,6 +201,32 @@ impl<T: Config> StakingLedger<T> {
 		}
 	}
 
+	/// Sets the ledger controller to its stash.
+	pub(crate) fn set_controller_to_stash(self) -> Result<(), Error<T>> {
+		let controller = self.controller.as_ref()
+            .defensive_proof("Ledger's controller field didn't exist. The controller should have been fetched using StakingLedger.")
+            .ok_or(Error::<T>::NotController)?;
+
+		ensure!(self.stash != *controller, Error::<T>::AlreadyPaired);
+
+		match Ledger::<T>::get(&self.stash) {
```

Comment:
Could use `if let Some()` syntax

Reply:
or actually if the key (stash) just exists in ledger that is probably enough to error. The last check `self.stash != controller` would cover the below conditions.